protocols/direct-fund/direct-fund.go

```diff
@@ -29,56 +29,40 @@ var ErrNotApproved = errors.New("objective not approved")
 
 // DirectFundObjective is a cache of data computed by reading from the store. It stores (potentially) infinite data
 type DirectFundObjective struct {
-	Status    protocols.ObjectiveStatus
-	channelId types.Destination
-
-	participantIndex map[types.Address]uint // the index for each participant
-	expectedStates   []state.State          // indexed by turn number
-
-	myIndex uint // my index in the Participants array of the initial state
-
-	preFundSigned []bool // indexed by participant.
+	Status protocols.ObjectiveStatus
+	C      *channel.Channel
 
 	myDepositSafetyThreshold types.Funds // if the on chain holdings are equal to this amount it is safe for me to deposit
 	myDepositTarget          types.Funds // I want to get the on chain holdings up to this much
 	fullyFundedThreshold     types.Funds // if the on chain holdings are equal
-
-	postFundSigned []bool // indexed by participant
-
-	onChainHolding types.Funds
 }
 
 // New initiates a DirectFundingInitialState with data calculated from
 // the supplied initialState and client address
-func New(initialState state.State, myAddress types.Address) (DirectFundObjective, error) {
+func New(initialState state.State, myAddress types.Address, isTwoPartyLedger bool, myDestination types.Destination, theirDestination types.Destination) (DirectFundObjective, error) {
 	if initialState.IsFinal {
 		return DirectFundObjective{}, errors.New("attempted to initiate new direct-funding objective with IsFinal == true")
 	}
 
-	var init DirectFundObjective
+	var init = DirectFundObjective{}
 	var err error
 
 	init.Status = protocols.Unapproved
-	init.channelId, err = initialState.ChannelId()
-	if err != nil {
-		return init, err
-	}
-	init.participantIndex = make(map[types.Address]uint)
-	for i, v := range initialState.Participants {
-		init.participantIndex[v] = uint(i)
-	}
-	init.expectedStates = make([]state.State, 2)
-	init.expectedStates[0] = initialState
-
-	init.expectedStates[1] = initialState.Clone()
-	init.expectedStates[1].TurnNum = big.NewInt(1)
 
+	var myIndex uint
 	for i, v := range initialState.Participants {
 		if v == myAddress {
-			init.myIndex = uint(i)
+			myIndex = uint(i)
 		}
 	}
 
+	init.C = &channel.Channel{}
+	*init.C, err = channel.New(initialState, isTwoPartyLedger, myIndex, myDestination, theirDestination)
+
+	if err != nil {
+		return init, err
```

Comment:
```suggestion
		return DirectFundObjective{}, fmt.Errorf("failed to initialize channel for direct-fund objective: %w", err)
```
- I'm preferring explicitly empty object returns in failure cases
- per a discussion w/ @andrewgordstewart yesterday, we should probably make a habit of wrapping errors from called-code in order to improve debuggability